core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -320,34 +419,35 @@ LookUp lookUp(K key) {
             }
             // if keys are equal - we've found the item
             else if (cmp == 0) {
-                int hi = getEntryField(curr, OFFSET_HANDLE_INDEX);
-                if (hi < 0) {
-                    return new LookUp(null, curr, -1);
+                long valueReference = getValueReference(curr);
+                Slice valueSlice = buildValueSlice(valueReference);
+                if (valueSlice == null) {
+                    assert valueReference == 0;
+                    return new LookUp(null, valueReference, curr);
                 }
-                Handle h = handles[hi];
-                if (h.isDeleted()) {
-                    return new LookUp(null, curr, hi);
+                if (ValueUtils.isValueDeleted(valueSlice)) {
+                    return new LookUp(null, valueReference, curr);
                 }
-                return new LookUp(h, curr, hi);
+                return new LookUp(valueSlice, valueReference, curr);
             }
             // otherwise- proceed to next item
             else {
-                curr = getEntryField(curr, OFFSET_NEXT);
+                curr = getEntryFieldInt(curr, OFFSET.NEXT);
             }
         }
         return null;
     }
 
     static class LookUp {
 
-        final Handle handle;
+        Slice valueSlice;
```

Comment:
For the sake of the future people who will work with this code. Please add a comments explaining that Slice actually is an object that references the value. Slice is needed for simplicity of accessing value. Also slice object equals null means the value is deleted. But valueReference is a serialized reference to a value compacted into long number. It is needed for future CAS if it will come.